projects/packages/boost-speed-score/src/class-speed-score.php

```diff
@@ -42,10 +42,19 @@ class Speed_Score {
 	/**
 	 * Constructor.
 	 *
-	 * @param Modules_Setup $modules - An instance of Automatic\Jetpack_Boost\Modules\Modules_Setup.
-	 * @param string        $client  - A string representing the client making the request.
+	 * @param string[] $modules - Array of module slugs that are currently active and can impact speed score.
+	 * @param string   $client  - A string representing the client making the request.
 	 */
 	public function __construct( $modules, $client ) {
+		/*
+		 * Plugins using the old version of the package may pass an object instead of an array.
+		 * Keeping it backward compatible by converting the object to an array.
+		 * We will loose the module slugs in case of an object, but it is better than a fatal error.
+		 */
+		if ( ! is_array( $modules ) ) {
+			$modules = array();
```

Comment:
Can we not just convert the object to an array?

Reply:
No. It's a complex object and as of stable, produces less accurate data anyway. Ones we release new version of boost, it will be coming in as an array and this will become redundant. I am hoping that the new release will go out next week with the updated lighthouse, so this won't be for long.